frontend/packages/client/src/hooks/useJoinCommunity.js

```diff
@@ -1,74 +1,77 @@
-import { useReducer, useCallback } from 'react';
-import { defaultReducer, INITIAL_STATE } from '../reducers';
+import { useMutation, useQueryClient } from 'react-query';
 import { useErrorHandlerContext } from '../contexts/ErrorHandler';
 import { getCompositeSigs } from 'utils';
 
 export default function useJoinCommunity() {
-  const [state, dispatch] = useReducer(defaultReducer, {
-    ...INITIAL_STATE,
-    loading: false,
-  });
+  const queryClient = useQueryClient();
   const { notifyError } = useErrorHandlerContext();
 
-  const createCommunityUser = useCallback(
-    async (communityId, user, injectedProvider) => {
-      const { addr } = user;
-      const { currentUser } = injectedProvider;
-      const { signUserMessage } = currentUser();
-      dispatch({ type: 'PROCESSING' });
-      const timestamp = Date.now().toString();
-      const hexTime = Buffer.from(timestamp).toString('hex');
-      const url = `${process.env.REACT_APP_BACK_END_SERVER_API}/communities/${communityId}/users`;
-      const _compositeSignatures = await signUserMessage(hexTime);
-      if (_compositeSignatures.indexOf('Declined:') > -1) {
-        dispatch({ type: 'SUCCESS' });
-        return { success: false };
-      }
-      const compositeSignatures = getCompositeSigs(_compositeSignatures);
-      if (!compositeSignatures) {
-        return { error: 'No valid user signature found.' };
-      }
+  const { mutateAsync: createCommunityUserMutation, mutate: mutateOne } =
+    useMutation(
```

Comment:
I had tu use `mutateAsync` as opposed to `mutate` because of the way the hook is used in the component. The component is expecting to get a result from the mutation like here: 

In the docs: https://react-query.tanstack.com/reference/useMutation  retuning the mutate function seems the way to go, they we could use something like:
```
const { mutate, data, error, isLoading } = useSomeWrapperFunctionMutation(mutate, 
{
   onSuccess: () => {
        manage side effects here
   }
   });
then we can call
mutate(payload);

or if we have specific side effects:
mutate(payload, {
   onSuccess: (data, variables, context) => {
     // I will fire second!
   },
   onError: (error, variables, context) => {
     // I will fire second!
   },
   onSettled: (data, error, variables, context) => {
     // I will fire second!
   },
 })